test/Miscellaneous/KeepContextInSuper.js

```diff
@@ -0,0 +1,43 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+WScript.LoadScriptFile("..\\UnitTestFramework\\UnitTestFramework.js");
+
+var tests = [
+    {
+        name: "The execution of the get or set attribute on an object should be \
+performed with the same context from which they were called from.",
+        body: function () {
+            var a = {};
+            class C {
+                // get and set should execute in the context of their caller, in
+                // this example that would be "a"
+                set f(x) { assert.isTrue(a === this); }
+                get f() {
+                    assert.isTrue(a === this);
+
+                    // Necessary as get must be called as a function and should
+                    // not throw an error
+                    return function () { }
+                }
+            }
+            class D extends C {
+                // The current context is "a" and not D since apply() and call()
+                // overrided D.
+                g() { super.f(); }
+                h() { super.f = 5; }
+                i() { super.f; }

```

Comment:
Please add some other super call types:
```javascript
let prop = 'f';
let prop2 = 'meth';
class C {
  // ...
  meth() { /* regular method validating 'this' */ }
}
class D extends C {
  // ...
  x() { return super['f']; }
  y() { return super[prop]; }
  z() { super['f'] = 'value'; }
  aa() { super[prop] = 'value'; }
  ab() { return super['meth'](); }
  ac() { return super[prop2](); }
}
```

Reply:
Thanks for the extra tests!

Tests using a property accessor with bracket notation (`super[prop]`) will result in a `LdElemI_A` opcode which cannot take another property to be used as the context during the execution of the property access - this limitation may require the addition of another opcode (`LdSuperElemI_A`)

Reply:
I am roughly aware of this. Shall we fix it as part of this PR?

Reply:
If you do, please also link #4835

Reply:
This PR will not fix these tests, though they have been added and commented out.